File: client/android/BigFoxAndroid/app/src/main/java/vn/com/onesoft/bigfox/io/message/core/cs/CSClientInfo.java
/*
 * Author: QuanPH
 * Copyright @ 2015 by OneSoft.,JSC
 * 
 */
package vn.com.onesoft.bigfox.io.message.core.cs;

import vn.com.onesoft.bigfox.io.core.BFUtils;
import vn.com.onesoft.bigfox.io.core.ConnectionManager;
import vn.com.onesoft.bigfox.io.core.BigFoxContext;
import vn.com.onesoft.bigfox.io.message.BaseMessage;
import vn.com.onesoft.bigfox.io.message.core.object.ClientInfo;
import vn.com.onesoft.bigfox.io.message.core.tags.CoreTags;
import vn.com.onesoft.bigfox.io.message.annotations.Message;
import vn.com.onesoft.bigfox.io.message.annotations.Property;

/**
 *
 * @author QuanPH
 */
@Message(tag = CoreTags.CS_CLIENT_INFO, name = "CS_CLIENT_INFO", isCore = true)
public class CSClientInfo extends BaseMessage {

    @Property(name = "clientInfo")
    private ClientInfo clientInfo;

    public CSClientInfo() {
        clientInfo = new ClientInfo();
        clientInfo.device = ClientInfo.DEVICE_ANDROID;
        clientInfo.imei = BigFoxContext.imei;

        if (ConnectionManager.getInstance().sessionId.length() == 0) {
            ConnectionManager.getInstance().sessionId = BFUtils.genRandomString(10);
        }
        clientInfo.sessionId = ConnectionManager.getInstance().sessionId;
        clientInfo.version = BigFoxContext.version;
        clientInfo.metadata = "";
    }

}

File: client/ios/BigFoxClient/BigFoxClient/message/core/cs/CSClientInfo.m
//
//  CSClientInfo.m
//  BigFoxClient
//
//  Created by QuyenNX on 9/29/15.
//  Copyright © 2015 QuyenNX. All rights reserved.
//

#import "CSClientInfo.h"
#import "ConnectionManager.h"
#import "BFUtils.h"
#import "BFConfig.h"
#import "CoreTags.h"


static int TAG = CS_CLIENT_INFO;
@implementation CSClientInfo
@synthesize clientInfo;

- (id) init {
    self = [super init:true];
    if (self) {
        self.tag = TAG;
        clientInfo = [[ClientInfo alloc] init];
        clientInfo.device = DEVICE_IOS;
        clientInfo.imei = @"";
        
        if ([ConnectionManager getInstance].sessionId.length == 0) {
            [ConnectionManager getInstance].sessionId = [BFUtils genRandStringLength:10];
        }
        clientInfo.sessionId = [[ConnectionManager getInstance] sessionId];
        clientInfo.version = VERSION;
        clientInfo.metadata= @"";
    }
    return  self;
}
@end
